Dockerfile

```diff
@@ -196,7 +196,7 @@ RUN rbenv install $RUBY_VERSION -v \
     && gem install bundler cocoapods:$COCOAPODS_VERSION
 
 FROM scratch AS ruby
-COPY --from=rubybuild $RBENV_ROOT $RBENV_ROOT
+COPY --from=rubybuild /opt/rbenv /opt/rbenv
```

Comment:
Except for the issue with the `DART_SDK` variable was any of those not working as intended? I would have expected that those variables were always defined in the respective parent image? I just want to understand what's the benefit of the change as it seems to me it's making maintenance harder.

Reply:
> Except for the issue with the `DART_SDK` variable was any of those not working as intended?

As I was trying to explain in the commit message, *all* of these lines were not working as intended:

- `FROM scratch AS ruby` starts a new stage with "scratch" (the empty image) as the parent. I.e. the parent image has no environment variables defined.
- `COPY --from=rubybuild $RBENV_ROOT $RBENV_ROOT` refers to "RBENV_ROOT" which is unset in this context. The "rubybuild" image used by `--from` *does* have "RBENV_ROOT" defined, but the source for a copy does not count as a parent / ancestor image.
- Thus the line expands to `COPY --from=rubybuild "" ""` which copies *everything*, not just the binaries from `/opt/rbenv`.

Reply:
BTW, I've verified this by looking at the generated layers / stages with [dive](https://github.com/wagoodman/dive).

Reply:
> BTW, I've verified this by looking at the generated layers / stages with [dive](https://github.com/wagoodman/dive).

Nice tool!

Reply:
Thanks for the explanation, now it makes sense to me.